File: medium/1657. Determine if Two Strings Are Close/main.dart
class Solution {
  bool closeStrings(String word1, String word2) {
    if (word1.length != word2.length) {
      return false;
    }

    List<int> firstFreq = List<int>.filled(26, 0);
    List<int> secondFreq = List<int>.filled(26, 0);

    for (int i = 0; i < word1.length; i++) {
      firstFreq[word1.codeUnitAt(i) - 'a'.codeUnitAt(0)]++;
      secondFreq[word2.codeUnitAt(i) - 'a'.codeUnitAt(0)]++;
    }

    for (int i = 0; i < 26; i++) {
      if ((firstFreq[i] == 0 && secondFreq[i] > 0) ||
          (firstFreq[i] == 0 && secondFreq[i] > 0)) {
        return false;
      }
    }

    firstFreq.sort();
    secondFreq.sort();

    for (int i = 0; i < 26; i++) {
      if (firstFreq[i] != secondFreq[i]) {
        return false;
      }
    }

    return true;
  }
}

File: medium/1657. Determine if Two Strings Are Close/main.rs
impl Solution {
    pub fn close_strings(word1: String, word2: String) -> bool {
        if word1.len() != word2.len() {
            return false;
        }

        let (mut first_freq, mut second_freq) = ([0; 26], [0; 26]);

        for c in word1.chars() {
            let idx = (c as u8 - b'a') as usize;
            first_freq[idx] += 1;
        }

        for c in word2.chars() {
            let idx = (c as u8 - b'a') as usize;
            second_freq[idx] += 1;
        }

        for i in 0..26 {
            if (first_freq[i] == 0 && second_freq[i] > 0)
                || (first_freq[i] > 0 && second_freq[i] == 0)
            {
                return false;
            }
        }

        first_freq.sort();
        second_freq.sort();

        first_freq == second_freq
    }
}
